x-pack/plugins/apm/common/projections/services.ts

```diff
@@ -7,37 +7,46 @@
 import {
   Setup,
   SetupUIFilters,
-  SetupTimeRange
+  SetupTimeRange,
+  SetupHasTransactionDurationMetrics
   // eslint-disable-next-line @kbn/eslint/no-restricted-paths
 } from '../../server/lib/helpers/setup_request';
-import { SERVICE_NAME, PROCESSOR_EVENT } from '../elasticsearch_fieldnames';
+import { SERVICE_NAME } from '../elasticsearch_fieldnames';
 // eslint-disable-next-line @kbn/eslint/no-restricted-paths
 import { rangeFilter } from '../../server/lib/helpers/range_filter';
 
 export function getServicesProjection({
   setup
 }: {
-  setup: Setup & SetupTimeRange & SetupUIFilters;
+  setup: Setup &
+    SetupTimeRange &
+    SetupUIFilters &
+    SetupHasTransactionDurationMetrics;
 }) {
-  const { start, end, uiFiltersES, indices } = setup;
+  const {
+    start,
+    end,
+    uiFiltersES,
+    indices,
+    hasTransactionDurationMetrics
+  } = setup;
+
+  const index = [
+    indices['apm_oss.metricsIndices'],
+    indices['apm_oss.errorIndices']
+  ];
+
+  if (!hasTransactionDurationMetrics) {
+    index.push(indices['apm_oss.transactionIndices']);
+  }
```

Comment:
I think it's worth drawing attention to this with some comments. Perhaps even duplicate the indicies if it makes it more clear:

```ts
const index = hasTransactionDurationMetrics
  ? // query pre-aggregated (histogram) docs
    [
      indices['apm_oss.metricsIndices'], 
      indices['apm_oss.errorIndices']
    ]
  : // query transaction docs
    [
      indices['apm_oss.metricsIndices'],
      indices['apm_oss.errorIndices'],
      indices['apm_oss.transactionIndices']
    ];
```